Formula/macaulay2.rb

```diff
@@ -0,0 +1,56 @@
+class Macaulay2 < Formula
+  @name = "M2"
+  desc "Software system for algebraic geometry research"
+  homepage "http://macaulay2.com"
+  url "https://github.com/Macaulay2/M2.git", :using => :git, :branch => "release-1.16"
+  version "1.16"
+  license "GPL-3.0"
+
+  head do
+    url "https://github.com/Macaulay2/M2.git", :using => :git, :branch => "development"
+  end
+
+  # autoconf, automake, and libtool are required to build
+  # libraries not available via brew, such as factory
```

Comment:
Those should really just be split out into different formulae

Reply:
That would have been great, but we're not the maintainers of most of them, and there's quite a few: factory, fflas_ffpack, givaro, mpsolve, 4ti2, cohomcalg, csdp, gfan, lrslib, nauty, normaliz, topcom.

Is this a blocking issue?

Reply:
Homebrew doesn't have dedicated maintainers for tools, we actually prefer if you didn't submit your own work.

As for the request, it's not a hard requirement but it is very much preferable and a lot cleaner to have all parts split out.

Reply:
For starters: there’s a formula for `nauty` on homebrew-core already.

Reply:
Thanks! This will probably take some time to do.